sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/SqlScriptingParserSuite.scala

```diff
@@ -161,6 +161,72 @@ class SqlScriptingParserSuite extends SparkFunSuite with SQLHelper {
       == "SELECT 3")
   }
 
+  test("compound: beginLabel") {
+    val batch =
+      """
+        |lbl: BEGIN
+        |  SELECT 1;
+        |  SELECT 2;
+        |  INSERT INTO A VALUES (a, b, 3);
+        |  SELECT a, b, c FROM T;
+        |  SELECT * FROM T;
+        |END""".stripMargin
+    val tree = parseScript(batch)
+    assert(tree.collection.length == 5)
+    assert(tree.collection.forall(_.isInstanceOf[SingleStatement]))
+    assert(tree.label.equals("lbl"))
+  }
+
+  test("compound: beginLabel + endLabel") {
+    val batch =
+      """
+        |lbl: BEGIN
+        |  SELECT 1;
+        |  SELECT 2;
+        |  INSERT INTO A VALUES (a, b, 3);
+        |  SELECT a, b, c FROM T;
+        |  SELECT * FROM T;
+        |END lbl""".stripMargin
+    val tree = parseScript(batch)
+    assert(tree.collection.length == 5)
+    assert(tree.collection.forall(_.isInstanceOf[SingleStatement]))
+    assert(tree.label.equals("lbl"))
+  }
+
+  test("compound: beginLabel + endLabel with different values") {
+    val batch =
+      """
+        |lbl_begin: BEGIN
+        |  SELECT 1;
+        |  SELECT 2;
+        |  INSERT INTO A VALUES (a, b, 3);
+        |  SELECT a, b, c FROM T;
+        |  SELECT * FROM T;
+        |END lbl_end""".stripMargin
+    val e = intercept[SparkException] {
+      parseScript(batch)
+    }
+    assert(e.getErrorClass === "INTERNAL_ERROR")
+    assert(e.getMessage.contains("Both labels should be same."))
+  }
+
+  test("compound: endLabel") {
+    val batch =
+      """
+        |BEGIN
+        |  SELECT 1;
+        |  SELECT 2;
+        |  INSERT INTO A VALUES (a, b, 3);
+        |  SELECT a, b, c FROM T;
+        |  SELECT * FROM T;
+        |END lbl""".stripMargin
+    val e = intercept[SparkException] {
+      parseScript(batch)
+    }
+    assert(e.getErrorClass === "INTERNAL_ERROR")
+    assert(e.getMessage.contains("End label can't exist without begin label."))
+  }
+
```

Comment:
Can you also add a test that verifies that unique name is given if label is not specified?

Reply:
Done